wave_matching/tests/icp_tests.cpp

```diff
@@ -0,0 +1,80 @@
+// Tests for wrapper
+
+#include <gtest/gtest.h>
+#include <pcl/io/pcd_io.h>
+#include <Eigen/Dense>
+#include <iostream>
+
+#include "wave/matching/ICP.hpp"
+
+bool isApprox(const Eigen::Affine3d& first,
+              const Eigen::Affine3d& second,
+              double TOL) {
+    for (uint8_t iii = 0; iii < 4; ++iii) {
+        for(uint8_t jjj = 0; jjj < 4; ++jjj) {
+            TOL -= std::abs(first(iii,jjj) - second(iii,jjj));
+        }
+    }
+    if (TOL > 0) {
```

Comment:
But it's not checking if it is equal to 0, just greater than. Shouldn't that work even with the floating point error?